ValveResourceFormat/IO/MaterialExtract.cs

```diff
@@ -0,0 +1,405 @@
+            {
+                if (texturePath.EndsWith(suffix, StringComparison.OrdinalIgnoreCase))
+                {
+                    texturePath = texturePath[..^suffix.Length];
+                    break;
+                }
+            }
+
+            texturePath += desiredSuffix;
+        }
+
+        if (IsDefaultTexture(texturePath))
+        {
+            return Path.ChangeExtension(texturePath, "tga");
+        }
+
+        return Path.ChangeExtension(texturePath, "png");
+    }
+
+    public IEnumerable<UnpackInfo> GetUnpackedTextures(string textureType, string texturePath,
+            Dictionary<string, long> intParams, Dictionary<string, Vector4> vectorParams)
+    {
+        var shader = material.ShaderName[..^4];
+
+        if (TextureMappings.ContainsKey(shader) && TextureMappings[shader] is IDictionary<string, ChannelMapping[]> mappings && mappings.ContainsKey(textureType))
+        {
+            // The first (file) input on the vtex decides the filename, so only it can keep the original suffix.
+            var keepOriginalSuffix = true;
+
+            foreach (var mapping in mappings[textureType])
+            {
+                var (channels, newTextureType) = mapping;
+
+                if (newTextureType.Length == 0 && !TryFigureOutNonStaticMap(shader, textureType, intParams, out newTextureType))
+                {
+                    continue;
+                }
+
+                if (vectorParams.ContainsKey(newTextureType))
+                {
+                    continue;
+                }
+
+                string desiredSuffix = null;
+                if (!keepOriginalSuffix)
+                {
+                    desiredSuffix = CommonTextureSuffixes.GetValueOrDefault(newTextureType, "-" + channels);
+                }
+
+                yield return new UnpackInfo
+                {
+                    TextureType = newTextureType,
+                    TextureFileName = OutTextureName(texturePath, desiredSuffix),
+                    Channels = channels
+                };
+
+                keepOriginalSuffix = false;
+            }
+
+            yield break;
+        }
+
+        // No pack info for this texture, so just clean up filename and try to guess its type.
+        var guessedTextureType = textureType.Replace("g_t", "Texture", StringComparison.Ordinal);
+        if (!vectorParams.ContainsKey(guessedTextureType))
+        {
+            Console.WriteLine($"Missing pack info for {textureType} in {shader}");
+            yield return new UnpackInfo
+            {
+                TextureType = guessedTextureType,
+                TextureFileName = OutTextureName(texturePath),
+                Channels = "RGBA"
+            };
+        }
+    }
+
+    public static bool TryFigureOutNonStaticMap(string shader, string textureType, Dictionary<string, long> intParams, out string newTextureType)
+    {
+        if (shader == "vr_simple" && textureType == "g_tColor")
```

Comment:
Writing shader specific code is oof. Any reason this data doesn't fit in the mapping above?